pytorch_lightning/callbacks/finetuning.py

```diff
@@ -27,31 +27,67 @@
 from pytorch_lightning import _logger as log
 from pytorch_lightning.callbacks.base import Callback
 from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 
 def multiplicative(epoch):
     return 2
 
 
-class BaseFinetuningCallback(Callback):
+class BaseFinetuning(Callback):
 
     r"""
-    BaseFinetuningCallback.
-    Overrides any functions with your own logic.
+    BaseFinetuning.
+    Override ``freeze_before_training`` and ``finetune_function`` with your own logic.
+
+    .. note:: Make sure to filter the parameters based on `requires_grad`
+
+    Example::
+
+        class MyModel(LightningModule)
+
+            ...
+
+            def configure_optimizer(self):
+                # Make sure to filter the parameters based on `requires_grad`
+                return Adam(filter(lambda p: p.requires_grad, parameters))
+
+        class FeatureExtractorFreezeUnfreeze(BaseFinetuning):
+
+            def __init__(self, unfreeze_at_epoch=10)
+                self._unfreeze_at_epoch = unfreeze_at_epoch
+
+            def freeze_before_training(self, pl_module):
+                # freeze any module you want
+                # Here, we are freezing ``feature_extractor``
+                self.freeze(pl_module.feature_extractor)
+
+            def finetune_function(self, pl_module, current_epoch, optimizer, opt_idx):
+                # When ``current_epoch`` is 10, feature_extractor will start to be trained.
+                if current_epoch == self._unfreeze_at_epoch:
+                    self.unfreeze_and_add_param_group(
+                        module=pl_module.feature_extractor,
+                        optimizer=optimizer,
+                        train_bn=True,
+                    )
     """
 
     BN_TYPES = (torch.nn.BatchNorm1d, torch.nn.BatchNorm2d, torch.nn.BatchNorm3d)
 
     @staticmethod
-    def _make_trainable(module: Module) -> None:
+    def _make_trainable(module: Union[Module, List[Module]]) -> None:
         """Unfreezes a given module.
         Args:
             module: The module to unfreeze
         """
-        for param in module.parameters():
-            param.requires_grad = True
-        module.train()
+        if isinstance(module, list):
+            for m in module:
+                BaseFinetuning._make_trainable(m)
```

Comment:
```suggestion
                self._make_trainable(m)
```

Reply:
everywhere?

Reply:
It is a staticmethod.

Reply:
we can access it using self. check this: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/gpu_stats_monitor.py